src/Microsoft.ApiDesignGuidelines.Analyzers/Core/DoNotDecreaseInheritedMemberVisibility.Fixer.cs

```diff
@@ -1,36 +1,88 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
-using System.Collections.Generic;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.CodeFixes;
-using Microsoft.CodeAnalysis.Editing;
 using System.Collections.Immutable;
 using System.Composition;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.CodeActions;
+using Analyzer.Utilities;
 
 namespace Microsoft.ApiDesignGuidelines.Analyzers
-{                              
+{
     /// <summary>
     /// CA2222: Do not decrease inherited member visibility
     /// </summary>
-    public abstract class DoNotDecreaseInheritedMemberVisibilityFixer : CodeFixProvider
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class DoNotDecreaseInheritedMemberVisibilityFixer : CodeFixProvider
     {
         public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(DoNotDecreaseInheritedMemberVisibilityAnalyzer.RuleId);
 
         public sealed override FixAllProvider GetFixAllProvider()
         {
-            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
             return WellKnownFixAllProviders.BatchFixer;
         }
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
-        {                              
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await new Task(() => { });
-            throw new NotImplementedException();
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var expression = root.FindNode(context.Span);
+
+            if (expression != null)
+            {
+                var title = MicrosoftApiDesignGuidelinesAnalyzersResources.DoNotDecreaseInheritedMemberVisibilityTitle;
+                context.RegisterCodeFix(CodeAction.Create(title, c => IncreaseVisibility(context.Document, expression, c), equivalenceKey: title), context.Diagnostics);
```

Comment:
Yes Title is what shows up in the lightbulb UI.

We log telemetry about which fix was invoked based on the typename of the codeaction. So if CodeAction.Create is used, it will have a generic name. So we derive from codeaction and have a nested type inside the fixer so that we can identify the fix. See other fixers in the solution for this pattern.